### PR TITLE
EventGrid: Support for patching Topic properties

### PR DESCRIPTION
**Pull request by kalyanaj**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR*   [*Note: There's one issue that appears to be a false positive - have sent out a mail regarding this.]

### CODE REVIEW

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-09-15-preview/EventGrid.json
Comment (dsgouda) [inline specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-09-15-preview/EventGrid.json]: Is there a chance that the service returns `200` status code? if yes the generated client might consider it to be an error response.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Other than the minor question looks okay.

**Comment by kalyanaj** [comment]
@dsgouda , thanks for the review. One question I have is that the autorest validation (https://travis-ci.org/Azure/azure-rest-api-specs/jobs/292903576) still is flagging the TrackedResourcePatchOperation error though patch support has been added for topics with this PR. Do you know why that could be happening?

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-09-15-preview/EventGrid.json
Comment (kalyanaj) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-09-15-preview/EventGrid.json]: Good point. No, the service always handles this asynchronously and always returns only 201, so I think we are fine leaving it with just 201 status code. Please let me know if that is fine.

**Comment by dsgouda** [comment]
I'll take a look at the validator errors

**Comment by dsgouda** [comment]
The error flagged is a bug in the validator which doesn't seem to consider the 201 responses, have opened an issue for this (here)[https://github.com/Azure/azure-openapi-validator/issues/109]

You can ignore this for now, I can verify that the spec is correct.

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-09-15-preview/EventGrid.json
Comment (dsgouda) [inline specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-09-15-preview/EventGrid.json]: nit: incorrect description

**Review by dsgouda — COMMENTED** [review]
Needs a minor description fix, otherwise LGTM

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-09-15-preview/EventGrid.json
Comment (kalyanaj) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-09-15-preview/EventGrid.json]: Will update the description to "Topic update request accepted.". I just want to double-check one thing: does this description I put here have to match with anything the service returns, or is this independent?

Comment (dsgouda) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-09-15-preview/EventGrid.json]: What the service returns must match the description since we also generate the documentation with this information, hope this answers your query

Comment (dsgouda) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-09-15-preview/EventGrid.json]: @kalyanaj Can you fix this so we can merge the PR soon?

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
CI needs to pass before I merge.

**Pull request merged**
